### PR TITLE
[stable/weave-cloud] use constant container name

### PR DESCRIPTION
**Pull request by errordeveloper**
#### What this PR does / why we need it:

  - fixes an issue with other tools that rely on constant container name

#### Checklist
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped

### CODE REVIEW

File: stable/weave-cloud/templates/deployment.yaml [hunk 1]
```diff
@@ -26,7 +26,7 @@ spec:
       {{- end }}
       terminationGracePeriodSeconds: 30
       containers:
-      - name: {{ .Values.agent.name }}
+      - name: agent
```
Comment (cpanato) [inline stable/weave-cloud/templates/deployment.yaml:29]: would say to keep the existing name `weave-agent` and then remove the variable from the values.yaml and in readme if exist.

but i did not fully understand why

File: stable/weave-cloud/Chart.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 appVersion: "1.1.0"
 name: weave-cloud
-version: 0.3.0
+version: 0.4.0
```
Comment (cpanato) [inline stable/weave-cloud/Chart.yaml]: should be 0.3.1

Comment (errordeveloper) [reply stable/weave-cloud/Chart.yaml]: ok.

File: stable/weave-cloud/templates/deployment.yaml (hunk 1 above)
Comment (errordeveloper) [reply stable/weave-cloud/templates/deployment.yaml:29]: The purpose of this PR is to make the container name match [the source](https://github.com/weaveworks/launcher/blob/a3573cef5c15771e2a11f6c582a16adc2b6ed2b8/service/static/agent.yaml.in#L68).

Comment (cpanato) [reply stable/weave-cloud/templates/deployment.yaml:29]: ok got it. thanks so much

**Pull request merged**